node/peer.js

```diff
@@ -95,6 +89,24 @@ function TChannelPeer(channel, hostPort, options) {
 
 inherits(TChannelPeer, EventEmitter);
 
+TChannelPeer.prototype.setPreferConnectionDirection = function setPreferConnectionDirection(direction) {
+    var self = this;
+    if (self.preferConnectionDirection === direction) {
+        return;
+    }
```

Comment:
Yes, that is commonly the case except for a few corner cases we have to support. For example, an advertisement request will initiate a peer as 'any' direction. If it happens to reach the exit node for the service, this peer should be changed to outgoing later.